docs/source/coding_guidelines.rst

```diff
@@ -0,0 +1,71 @@
+Coding Style Guidelines
+=======================
+
+DataTransferKit developers follow a set of style guidelines and use the clang
+format tool to create a consistent appearance to all source code committed to
+the repository.
+
+ClangFormat
+-----------
+
+ClangFormat (version 3.9) is used to check the C++ code formatting style in
+DTK.  A pull request that does not comply will be rejected. Configure with
+``-D DataTransferKit_ENABLE_ClangFormat=ON`` and do ``make format-cpp`` to
+apply the formatting style before your commit.  Alternatively, run
+``ctest -V -R check_format_cpp`` display the diff without applying the
+changes.
+
+Style Guide
+-----------
+
+The following conventions are used in the code.
+
+Names of classes, structs, and enumerations are camel case and capitalized:
+
+.. code-block:: c++
+
+   class ExampleClassName {};
+
+Function names are camel case and not capitalized:
+
+.. code-block:: c++
+
+   void exampleFunctionName() const;
+
+Variable names are lower case and have underscores to separate words:
+
+.. code-block:: c++
+
+   double example_double_var;
+   std::vector<int> example_vec_var;
+
+If a variable is also class data that is private, protected, or public it is
+prefixed with a ``d_``:
```

Comment:
I am not a fan of d_ either but I am strongly against changing the convention. It's not like the indentation script where if something breaks it would only look bad. We need to write a new script to change the convention and if there is a corner case that we didn't think about, the code won't compile (if we are lucky) or we will introduce a new bug. I don't think it's worth the risk to change the convention.